views/js/component/activityMonitoring/activityGraph.js

```diff
@@ -0,0 +1,134 @@
+            padding: {
+                bottom: 0,
+                left: 35,
+            },
+            data: {
+                x: 'time',
+                xFormat: '%Y-%m-%d %H:%M:%S',
+                mimeType: 'json',
+                type: 'bar'
+            },
+            axis: {
+                x: {
+                    type: 'timeseries',
+                    tick: {
+                        format: '%H:%M',
+                    },
+                    label: {
+                        position: 'bottom center'
+                    }
+                },
+                y: {
+                    label: {
+                        position: 'top'
+                    }
+                }
+            },
+        }
+    };
+
+    /**
+     * @param {jQuery} $refreshProgress
+     * @param duration
+     */
+    function runRefreshProgressBar($refreshProgress, duration) {
+        $refreshProgress.stop();
+        $refreshProgress.css({width:'0%'});
+        $refreshProgress.animate({
+            width: '100%'
+        }, duration, 'linear');
+    }
+
+    /**
+     * Creates a dates range with date pickers
+     *
+     * @param {Object} config
+     * @param {String} [config.graphConfig] - configuration of c3 chart
+     * @param {String} [config.autoRefresh] - interval of auto refresh
+     * @param {String} [config.autoRefreshBar] - show auto refresh bar
+     */
+    function activityGraphFactory(config) {
+        var initConfig = _.merge({}, _defaults, config);
+        var chart;
+        var $refreshProgress;
+        var activityGraph = {
+            /**
+             * Refresh the graph
+             * @param {Object} [params]
+             */
+            refresh: function refresh() {
+                if (chart) {
+                    chart.load();
+                }
+            }
+        };
+
+        return component(activityGraph)
+            .on('render', function() {
+                chart = c3.generate(initConfig.graphConfig);
+                if (initConfig.autoRefresh) {
+                    $(initConfig.graphConfig.bindto).after($(
+                        '<div class="js-completed-assessments-refresh-bar refresh-bar"><div class="js-refresh-bar-progress refresh-bar-progress"></div></div>'
+                    ));
+                    $refreshProgress = $('.js-refresh-bar-progress');
+
+                    if (initConfig.autoRefreshBar) {
+                        runRefreshProgressBar($refreshProgress, initConfig.autoRefresh);
+                    }
+                    setInterval(function() {
+                        chart.load(initConfig.graphConfig.data);
+                        if (initConfig.autoRefreshBar) {
+                            runRefreshProgressBar($refreshProgress, initConfig.autoRefresh);
+                        }
+                    }, initConfig.autoRefresh);
+                }
+            })
+            .init(initConfig);
+    }
+
+    return activityGraphFactory;
```

Comment:
Would be nice to add an unit test to this component to verify the lifecycle of this component thoroughly.  External libs such c3 can be mocked to test this component individually. Many examples exist under tao/test/ui as examples.